### PR TITLE
Add notification API tests

### PR DESCRIPTION
**Pull request by d-rita**
This PR adds tests related to the Notifications API endpoints. 

**Related issue:**
- #5442

**How to test:**
- create a test database following the instructions [here](https://github.com/hotosm/tasking-manager/blob/develop/docs-old/setup-development.md#create-the-database-user-and-database). The database name is preceded by `test_`
- pull and checkout test branch using `git fetch origin && git checkout chore/add-notifications-tests`
- run tests using `python3 -m unittest discover tests/backend`

### CODE REVIEW

**Comment by Aadesh-Baral** [comment]
@d-rita is this ready for review?

**Comment by d-rita** [comment]
> @d-rita is this ready for review?

@Aadesh-Baral not yet, two failing tests for the get and post unread count api endpoints

File: tests/backend/integration/api/notifications/test_resources.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    def test_get_message_notifications_returns_401(self):
+        """
+        Test that endpoint returns 401 when an unauthenticated user accesses messsage notifications
+        """
+        response = self.client.get(self.url)
+        response_body = response.get_json()
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response_body["SubCode"], "InvalidToken")
+
+    def test_get_messages_returns_200(self):
```
Comment (Aadesh-Baral) [inline tests/backend/integration/api/notifications/test_resources.py:146]: Also, test all of the query parameters to ensure that they all have the expected impact on the response.

**Comment by Aadesh-Baral** [comment]
@d-rita What's the status of this PR?

**Comment by d-rita** [comment]
@Aadesh-Baral this is ready for review.
- I have included tests for the query parameters attached to the NotificationsAllApi.
- I have also updated the message_type filter implementation for a user to enter a string value of the message type, i.e. `/api/v2/notifications/?messageType=system`

**Review by Aadesh-Baral — COMMENTED** [review]
Test cases for query param sortBy and sortDirection seems to be missing for NotificationsAllAPI. Other than that LGTM.

File: backend/services/messaging/message_service.py [hunk 2]
```diff
@@ -706,8 +706,17 @@ def get_all_messages(
...
+                key.strip().lower() for key in message_type.split(",")
+            ]
+
+            filter_values = [
+                key.value
+                for key in MessageType
+                if key.name.lower() in message_type_filters
+            ]
+
+            query = query.filter(Message.message_type.in_(filter_values))
```
Comment (Aadesh-Baral) [inline backend/services/messaging/message_service.py:719]: Can you explain this change?

Comment (d-rita) [reply backend/services/messaging/message_service.py:719]: While testing, I could not retrieve any notifications if I specified the message type. Not when I used numbers (for the mapped integer value assigned to a type under the MessageType enum class), e.g. 1 (system), 2 (broadcast)., etc. Nor when I typed in the string value itself, e.g. system/SYSTEM/System. That was with the previous implementation.

So, this new implementation lets a user retrieve the specific messages when they filter by the string value of the message type, e.g. system, mention_notification, broadcast, etc. 

Could you test to confirm if my findings were accurate or not? @Aadesh-Baral

Comment (Aadesh-Baral) [reply backend/services/messaging/message_service.py:719]: This change has introduced a problem on the frontend. In our frontend code, we have four notification types that are associated with different combinations of MessageTypes. You can find more information about this issue in the following link: [GitHub Issue Comment](https://github.com/hotosm/tasking-manager/pull/5676#issuecomment-1506579190).

For instance, when applying a teams filter in the frontend, a query parameter is passed as "types=6,7,11" in the NotificationsAllAPI. To filter and retrieve the notifications for the provided message types on the backend, this value needs to be split by comma (',') and converted to integers.

However, due to this change, the value is being split correctly but is not being converted to integers, resulting in empty results being returned.

Comment (d-rita) [reply backend/services/messaging/message_service.py]: Noted. I have adjusted the tests, removed this change, and added tests for the sortBy and sortDirection query parameters. 
Sidenote: For future reference, we should probably include in the documentation the expected values for the message_type filter and what type the respective numerical values correspond to.

Comment (Aadesh-Baral) [reply backend/services/messaging/message_service.py]: > For future reference, we should probably include in the documentation the expected values for the message_type filter and what type the respective numerical values correspond to.

Sounds good

Comment (Aadesh-Baral) [reply backend/services/messaging/message_service.py]: > For future reference, we should probably include in the documentation the expected values for the message_type filter and what type the respective numerical values correspond to.

@d-rita Let's do this in this PR itself.

**Review by Aadesh-Baral — APPROVED** [review]

**Pull request merged**